### DISCUSSION TITLE
Multiple service calls from an Effect

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#367] [Announcements]
# Multiple service calls from an Effect

A snippet on how to invoke multiple service calls within a single NgRx Effect.

https://timdeschryver.dev/blog/multiple-service-calls-from-an-effect

**Comment by goldsam** [comment]
# Multiple actions from a single effect

Doesn't this violate the "one action per effect" best practice? 

##  Dependent fetches
On a related topic, what do you consider the best way to deal with dependent fetches? 

1. I would argue that using a chain of actions is a poor choice if the dependencies are not needed in every context. I could distinguish between these contexts with XLoadedSucessfullyInContextY, but this doesn't scale well as additional contexts are introduced. In fact, this touches on my criticism of "Good Action Hygene" - it flagrantly violates the Open/Closed principle. An alternative would be to use a parameterized initial action (with perhaps a `needsDependencies` property) which gets propagated along to the dependency actions

2. Handle all the fetches and store updates in a single effect and just dispatch actions for all the successful API fetches. I already have other code which relies on these "actions" (really events). I don't want to keep violating Open/Closed and events are a classic approach to building extensible systems.
